### PR TITLE
dev-util/gammaray: new package

### PR DESCRIPTION
**Pull request by Petross404**
High-level runtime introspection tool for Qt applications

Package-Manager: Portage-2.3.36, Repoman-2.3.9

### CODE REVIEW

File: dev-util/gammaray/gammaray-2.9.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Copyright 1999-2018 Gentoo Foundation
```
Comment (mgorny) [inline dev-util/gammaray/gammaray-2.9.0.ebuild]: Could you please update this PR to 2.9.1?

File: dev-util/gammaray/gammaray-2.9.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+if [[ ${PV} == 9999 ]] ; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/KDAB/GammaRay.git"
+else
+	SRC_URI="https://github.com/KDAB/GammaRay/releases/download/v${PV}/${P}.tar.gz"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="GPL-2 BSD MIT"
```
Comment (mgorny) [inline dev-util/gammaray/gammaray-2.9.0.ebuild]: README says it's GPL-2+.

How are the other licenses used? I see that they come from some bundled libraries. Is your ebuild suffering from them?

Comment (Petross404) [reply dev-util/gammaray/gammaray-2.9.0.ebuild]: I will take a look at this in a few days.

Comment (Petross404) [reply dev-util/gammaray/gammaray-2.9.0.ebuild]: @mgorny 
>README says it's GPL-2+.

I really can't remember why asturm proposed to include these too.

Comment (mgorny) [reply dev-util/gammaray/gammaray-2.9.0.ebuild]: @a17r ?

Comment (a17r) [reply dev-util/gammaray/gammaray-2.9.0.ebuild]: I didn't look at the code myself, iirc you found those bundled libs in use were having these other licences.

Comment (Petross404) [reply dev-util/gammaray/gammaray-2.9.0.ebuild]: @mgorny @a17r 

Should we leave the LICENSE as is?

Comment (mgorny) [reply dev-util/gammaray/gammaray-2.9.0.ebuild]: I reread my own question now. I'm sorry that I was unclear. I was asking if the bundled libraries are actually being used, and what's the chance of using system libraries instead.

Comment (Petross404) [reply dev-util/gammaray/gammaray-2.9.0.ebuild]: @mgorny No worries. I am afraid I can't check this until Monday, but AFAIK `add_subdirectory(3rdparty/*` doesn't contain lz4 and the other directories can be disabled*.

*
```
    sed -i "/BackwardMacros.cmake/d" CMakeLists.txt || die
    sed -i "/add_backward(gammaray_core)/d" core/CMakeLists.txt || die
```
and 
```
		-DGAMMARAY_BUILD_UI=ON
		-DGAMMARAY_DISABLE_FEEDBACK=ON
```
are doing this thing.

Comment (mgorny) [reply dev-util/gammaray/gammaray-2.9.0.ebuild]: Sure, I don't think we are in a hurry here, are we?

File: dev-util/gammaray/gammaray-2.9.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="GPL-2+"
+SLOT=0
+IUSE="3d bluetooth concurrent designer doc geolocation printsupport
+	script scxml svg test qml wayland webengine"
+
+RDEPEND="
+	$(add_qt_dep qtcore)
```
Comment (mgorny) [inline dev-util/gammaray/gammaray-2.9.1.ebuild]: Didn't I complain about using this horrible eclass already? So many subshell calls are going to have serious impact on metadata cache generation speed, and I see no apparent reason to use magic to generate dependency strings. Especially given that having the proper cat/pn would be much more readable here.

Comment (a17r) [reply dev-util/gammaray/gammaray-2.9.1.ebuild]: If it's not an actual KDE hosted project, I agree it does not make much sense to inherit kde5-functions just for the dep convenience.

Comment (Petross404) [reply dev-util/gammaray/gammaray-2.9.1.ebuild]: Good I will remove this.

File: dev-util/gammaray/gammaray-2.9.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="GPL-2+"
+SLOT=0
+IUSE="3d bluetooth concurrent designer doc geolocation printsupport
+	script scxml svg test qml wayland webengine"
+
+RDEPEND="
+	dev-qt/qtcore
```
Comment (a17r) [inline dev-util/gammaray/gammaray-2.9.1.ebuild]: Not quite, you still need to depend on `SLOT 5.

Comment (Petross404) [reply dev-util/gammaray/gammaray-2.9.1.ebuild]: Hmm, I suppose you are right. Removing Qt4 isn't enough and leaving Qt5 only isn't enough to be safe.

**Comment by mgorny** [comment]
Thanks. I'm going to test it now and merge if it builds/tests fine.

**Comment by mgorny** [comment]
Would you expect it to fail to build? ;-)

```
[492/882] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++-7.3.0 -DQT_CORE_LIB -DQT_DISABLE_DEPRECATED_BEFORE=0x040800 -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_NO_URL_CAST_FROM_STRING -DQT_STRICT_ITERATORS -DQT_TESTCASE_BUILDDIR=\"/tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1_build\" -DQT_TESTLIB_LIB -DQT_USE_FAST_CONCATENATION -DQT_USE_FAST_OPERATOR_PLUS -DQT_WIDGETS_LIB -Itests -I/tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1/tests -Itests/launcheruiiptest_autogen/include -I/tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1 -I/tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1/3rdparty -I. -Icommon -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtTest -isystem /usr/include/qt5/QtCore -isystem /usr/lib64/qt5/mkspecs/linux-g++ -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtNetwork  -DNDEBUG -march=k8-sse3 -mcx16 -msahf --param l1-cache-size=64 --param l1-cache-line-size=64 --param l2-cache-size=512 -O2 -pipe -frecord-gcc-switches -Wunused-but-set-variable -Wlogical-op -Wsizeof-pointer-memaccess -Wreorder -Wformat-security -Wno-c++1z-extensions -Wall -Wextra -pedantic -Woverloaded-virtual -Winit-self -Wmissing-include-dirs -Wunused -Wundef -Wpointer-arith -Wmissing-noreturn -Werror=return-type -Wswitch -std=gnu++0x -fno-operator-names -Wzero-as-null-pointer-constant -fvisibility=hidden -fvisibility-inlines-hidden   -fPIC -std=gnu++11 -MD -MT tests/CMakeFiles/launcheruiiptest.dir/launcheruiiptest.cpp.o -MF tests/CMakeFiles/launcheruiiptest.dir/launcheruiiptest.cpp.o.d -o tests/CMakeFiles/launcheruiiptest.dir/launcheruiiptest.cpp.o -c /tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1/tests/launcheruiiptest.cpp
FAILED: tests/CMakeFiles/launcheruiiptest.dir/launcheruiiptest.cpp.o 
/usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++-7.3.0 -DQT_CORE_LIB -DQT_DISABLE_DEPRECATED_BEFORE=0x040800 -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_NO_URL_CAST_FROM_STRING -DQT_STRICT_ITERATORS -DQT_TESTCASE_BUILDDIR=\"/tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1_build\" -DQT_TESTLIB_LIB -DQT_USE_FAST_CONCATENATION -DQT_USE_FAST_OPERATOR_PLUS -DQT_WIDGETS_LIB -Itests -I/tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1/tests -Itests/launcheruiiptest_autogen/include -I/tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1 -I/tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1/3rdparty -I. -Icommon -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtTest -isystem /usr/include/qt5/QtCore -isystem /usr/lib64/qt5/mkspecs/linux-g++ -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtNetwork  -DNDEBUG -march=k8-sse3 -mcx16 -msahf --param l1-cache-size=64 --param l1-cache-line-size=64 --param l2-cache-size=512 -O2 -pipe -frecord-gcc-switches -Wunused-but-set-variable -Wlogical-op -Wsizeof-pointer-memaccess -Wreorder -Wformat-security -Wno-c++1z-extensions -Wall -Wextra -pedantic -Woverloaded-virtual -Winit-self -Wmissing-include-dirs -Wunused -Wundef -Wpointer-arith -Wmissing-noreturn -Werror=return-type -Wswitch -std=gnu++0x -fno-operator-names -Wzero-as-null-pointer-constant -fvisibility=hidden -fvisibility-inlines-hidden   -fPIC -std=gnu++11 -MD -MT tests/CMakeFiles/launcheruiiptest.dir/launcheruiiptest.cpp.o -MF tests/CMakeFiles/launcheruiiptest.dir/launcheruiiptest.cpp.o.d -o tests/CMakeFiles/launcheruiiptest.dir/launcheruiiptest.cpp.o -c /tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1/tests/launcheruiiptest.cpp
distcc[7101] ERROR: compile (null) on localhost failed
In file included from /tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1/tests/launcheruiiptest.cpp:29:0:
/tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1/launcher/ui/connectpage.h:32:10: fatal error: gammaray_launcher_ui_export.h: No such file or directory
 #include "gammaray_launcher_ui_export.h"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
distcc[7100] ERROR: compile /tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1/tests/launcheruiiptest.cpp on pomiocik.local/4 failed

```

[dev-util:gammaray-2.9.1:20180915-180122.log](https://github.com/gentoo/gentoo/files/2385999/dev-util.gammaray-2.9.1.20180915-180122.log)

**Comment by Petross404** [comment]
> 
> 
> Would you expect it to fail to build? ;-)
> 
> ```
> [492/882] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++-7.3.0 -DQT_CORE_LIB -DQT_DISABLE_DEPRECATED_BEFORE=0x040800 -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_NO_URL_CAST_FROM_STRING -DQT_STRICT_ITERATORS -DQT_TESTCASE_BUILDDIR=\"/tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1_build\" -DQT_TESTLIB_LIB -DQT_USE_FAST_CONCATENATION -DQT_USE_FAST_OPERATOR_PLUS -DQT_WIDGETS_LIB -Itests -I/tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1/tests -Itests/launcheruiiptest_autogen/include -I/tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1 -I/tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1/3rdparty -I. -Icommon -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtTest -isystem /usr/include/qt5/QtCore -isystem /usr/lib64/qt5/mkspecs/linux-g++ -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtNetwork  -DNDEBUG -march=k8-sse3 -mcx16 -msahf --param l1-cache-size=64 --param l1-cache-line-size=64 --param l2-cache-size=512 -O2 -pipe -frecord-gcc-switches -Wunused-but-set-variable -Wlogical-op -Wsizeof-pointer-memaccess -Wreorder -Wformat-security -Wno-c++1z-extensions -Wall -Wextra -pedantic -Woverloaded-virtual -Winit-self -Wmissing-include-dirs -Wunused -Wundef -Wpointer-arith -Wmissing-noreturn -Werror=return-type -Wswitch -std=gnu++0x -fno-operator-names -Wzero-as-null-pointer-constant -fvisibility=hidden -fvisibility-inlines-hidden   -fPIC -std=gnu++11 -MD -MT tests/CMakeFiles/launcheruiiptest.dir/launcheruiiptest.cpp.o -MF tests/CMakeFiles/launcheruiiptest.dir/launcheruiiptest.cpp.o.d -o tests/CMakeFiles/launcheruiiptest.dir/launcheruiiptest.cpp.o -c /tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1/tests/launcheruiiptest.cpp
> FAILED: tests/CMakeFiles/launcheruiiptest.dir/launcheruiiptest.cpp.o 
> /usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++-7.3.0 -DQT_CORE_LIB -DQT_DISABLE_DEPRECATED_BEFORE=0x040800 -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_NO_URL_CAST_FROM_STRING -DQT_STRICT_ITERATORS -DQT_TESTCASE_BUILDDIR=\"/tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1_build\" -DQT_TESTLIB_LIB -DQT_USE_FAST_CONCATENATION -DQT_USE_FAST_OPERATOR_PLUS -DQT_WIDGETS_LIB -Itests -I/tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1/tests -Itests/launcheruiiptest_autogen/include -I/tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1 -I/tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1/3rdparty -I. -Icommon -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtTest -isystem /usr/include/qt5/QtCore -isystem /usr/lib64/qt5/mkspecs/linux-g++ -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtNetwork  -DNDEBUG -march=k8-sse3 -mcx16 -msahf --param l1-cache-size=64 --param l1-cache-line-size=64 --param l2-cache-size=512 -O2 -pipe -frecord-gcc-switches -Wunused-but-set-variable -Wlogical-op -Wsizeof-pointer-memaccess -Wreorder -Wformat-security -Wno-c++1z-extensions -Wall -Wextra -pedantic -Woverloaded-virtual -Winit-self -Wmissing-include-dirs -Wunused -Wundef -Wpointer-arith -Wmissing-noreturn -Werror=return-type -Wswitch -std=gnu++0x -fno-operator-names -Wzero-as-null-pointer-constant -fvisibility=hidden -fvisibility-inlines-hidden   -fPIC -std=gnu++11 -MD -MT tests/CMakeFiles/launcheruiiptest.dir/launcheruiiptest.cpp.o -MF tests/CMakeFiles/launcheruiiptest.dir/launcheruiiptest.cpp.o.d -o tests/CMakeFiles/launcheruiiptest.dir/launcheruiiptest.cpp.o -c /tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1/tests/launcheruiiptest.cpp
> distcc[7101] ERROR: compile (null) on localhost failed
> In file included from /tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1/tests/launcheruiiptest.cpp:29:0:
> /tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1/launcher/ui/connectpage.h:32:10: fatal error: gammaray_launcher_ui_export.h: No such file or directory
>  #include "gammaray_launcher_ui_export.h"
>           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> compilation terminated.
> distcc[7100] ERROR: compile /tmp/portage/dev-util/gammaray-2.9.1/work/gammaray-2.9.1/tests/launcheruiiptest.cpp on pomiocik.local/4 failed
> ```
> 
> [dev-util:gammaray-2.9.1:20180915-180122.log](https://github.com/gentoo/gentoo/files/2385999/dev-util.gammaray-2.9.1.20180915-180122.log)
/
Any case distcc is the problem?

**Comment by mgorny** [comment]
Doubt it. More likely it's race condition in parallel make rules (i.e. they failed to specify the deps between objects).

**Comment by Petross404** [comment]
Whatever the case I have already reported it.

**Comment by mgorny** [comment]
Maybe check if there's a separate target to build all those .h files early.

**Comment by Petross404** [comment]
> I won't be able as I don't have access to my pc for some days.

**Comment by mgorny** [comment]
No worry, I'll try to look at it when I finish upgrading Chromium, i.e. in 1-2 days ;-).

**Comment by Petross404** [comment]
Let me know if this fails to build again.

**Comment by mgorny** [comment]
'fraid so:

```
FAILED: tests/CMakeFiles/launcheruiiptest.dir/launcheruiiptest.cpp.o 
/usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++-8.3.0 -DQT_CORE_LIB -DQT_DISABLE_DEPRECATED_BEFORE=0x040800 -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_NO_URL_CAST_FROM_STRING -DQT_STRICT_ITERATORS -DQT_TESTCASE_BUILDDIR=\"/tmp/portage/dev-util/gammaray-2.10.0/work/gammaray-2.10.0_build\" -DQT_TESTLIB_LIB -DQT_USE_FAST_CONCATENATION -DQT_USE_FAST_OPERATOR_PLUS -DQT_WIDGETS_LIB -Itests -I/tmp/portage/dev-util/gammaray-2.10.0/work/gammaray-2.10.0/tests -Itests/launcheruiiptest_autogen/include -I/tmp/portage/dev-util/gammaray-2.10.0/work/gammaray-2.10.0 -I/tmp/portage/dev-util/gammaray-2.10.0/work/gammaray-2.10.0/3rdparty -I. -Icommon -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtTest -isystem /usr/include/qt5/QtCore -isystem /usr/lib64/qt5/mkspecs/linux-g++ -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtNetwork  -DNDEBUG -march=x86-64 -mtune=k8 -mcx16 -msahf -msse3 --param l1-cache-size=64 --param l1-cache-line-size=64 --param l2-cache-size=512 -O2 -pipe -frecord-gcc-switches -Wunused-but-set-variable -Wlogical-op -Wsizeof-pointer-memaccess -Wreorder -Wformat-security -Wall -Wextra -pedantic -Woverloaded-virtual -Winit-self -Wmissing-include-dirs -Wunused -Wundef -Wpointer-arith -Wmissing-noreturn -Werror=return-type -Wswitch -std=gnu++0x -fno-operator-names -Wzero-as-null-pointer-constant -fvisibility=hidden -fvisibility-inlines-hidden   -fPIC -std=gnu++11 -MD -MT tests/CMakeFiles/launcheruiiptest.dir/launcheruiiptest.cpp.o -MF tests/CMakeFiles/launcheruiiptest.dir/launcheruiiptest.cpp.o.d -o tests/CMakeFiles/launcheruiiptest.dir/launcheruiiptest.cpp.o -c /tmp/portage/dev-util/gammaray-2.10.0/work/gammaray-2.10.0/tests/launcheruiiptest.cpp
distcc[1080] ERROR: compile (null) on localhost failed
In file included from /tmp/portage/dev-util/gammaray-2.10.0/work/gammaray-2.10.0/tests/launcheruiiptest.cpp:29:
/tmp/portage/dev-util/gammaray-2.10.0/work/gammaray-2.10.0/launcher/ui/connectpage.h:32:10: fatal error: gammaray_launcher_ui_export.h: No such file or directory
 #include "gammaray_launcher_ui_export.h"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```
[dev-util:gammaray-2.10.0:20190403-072755.log](https://github.com/gentoo/gentoo/files/3037417/dev-util.gammaray-2.10.0.20190403-072755.log)

**Comment by juippis** [comment]
Ping.

**Comment by a17r** [comment]
ping. it would be sad to throw away this good work.

**Comment by Petross404** [comment]
Is it still failing? I asked once again but I don't have high hopes.

**Comment by juippis** [comment]
Yes, 
```
/tmp/portage/dev-util/gammaray-2.10.0/work/gammaray-2.10.0/core/probe.cpp: In constructor 'GammaRay::Probe::Probe(QObject*)':
/tmp/portage/dev-util/gammaray-2.10.0/work/gammaray-2.10.0/core/probe.cpp:255:38: error: 'class QBasicAtomicPointer<QSignalSpyCallbackSet>' has no member named 'signal_begin_callback'
  255 |         = qt_signal_spy_callback_set.signal_begin_callback;
      |                                      ^~~~~~~~~~~~~~~~~~~~~
/tmp/portage/dev-util/gammaray-2.10.0/work/gammaray-2.10.0/core/probe.cpp:257:38: error: 'class QBasicAtomicPointer<QSignalSpyCallbackSet>' has no member named 'signal_end_callback'
  257 |         = qt_signal_spy_callback_set.signal_end_callback;
      |                                      ^~~~~~~~~~~~~~~~~~~
/tmp/portage/dev-util/gammaray-2.10.0/work/gammaray-2.10.0/core/probe.cpp:259:38: error: 'class QBasicAtomicPointer<QSignalSpyCallbackSet>' has no member named 'slot_begin_callback'
  259 |         = qt_signal_spy_callback_set.slot_begin_callback;
      |                                      ^~~~~~~~~~~~~~~~~~~
/tmp/portage/dev-util/gammaray-2.10.0/work/gammaray-2.10.0/core/probe.cpp:260:81: error: 'class QBasicAtomicPointer<QSignalSpyCallbackSet>' has no member named 'slot_end_callback'
  260 | sSignalSpyCallbackSet.slotEndCallback = qt_signal_spy_callback_set.slot_end_callback;
      |                                                                    ^~~~~~~~~~~~~~~~~

/tmp/portage/dev-util/gammaray-2.10.0/work/gammaray-2.10.0/core/probe.cpp: In destructor 'virtual GammaRay::Probe::~Probe()':
/tmp/portage/dev-util/gammaray-2.10.0/work/gammaray-2.10.0/core/probe.cpp:279:38: error: cannot convert 'const QSignalSpyCallbackSet' to 'QSignalSpyCallbackSet*'
  279 |     qt_register_signal_spy_callbacks(prevCallbacks);
      |                                      ^~~~~~~~~~~~~
      |                                      |
      |                                      const QSignalSpyCallbackSet
In file included from /tmp/portage/dev-util/gammaray-2.10.0/work/gammaray-2.10.0/core/probe.cpp:74:
/usr/include/qt5/QtCore/5.14.1/QtCore/private/qobject_p.h:82:76: note:   initializing argument 1 of 'void qt_register_signal_spy_callbacks(QSignalSpyCallbackSet*)'
   82 |  Q_CORE_EXPORT qt_register_signal_spy_callbacks(QSignalSpyCallbackSet *callback_set);
      |                                                 ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~

/tmp/portage/dev-util/gammaray-2.10.0/work/gammaray-2.10.0/core/probe.cpp: In member function 'void GammaRay::Probe::setupSignalSpyCallbacks()':
/tmp/portage/dev-util/gammaray-2.10.0/work/gammaray-2.10.0/core/probe.cpp:1022:38: error: cannot convert 'QSignalSpyCallbackSet' to 'QSignalSpyCallbackSet*'
 1022 |     qt_register_signal_spy_callbacks(cbs);
      |                                      ^~~
      |                                      |
      |                                      QSignalSpyCallbackSet
In file included from /tmp/portage/dev-util/gammaray-2.10.0/work/gammaray-2.10.0/core/probe.cpp:74:
/usr/include/qt5/QtCore/5.14.1/QtCore/private/qobject_p.h:82:76: note:   initializing argument 1 of 'void qt_register_signal_spy_callbacks(QSignalSpyCallbackSet*)'
   82 |  Q_CORE_EXPORT qt_register_signal_spy_callbacks(QSignalSpyCallbackSet *callback_set);
      |                                                 ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~

make[2]: *** [core/CMakeFiles/gammaray_core.dir/build.make:200: core/CMakeFiles/gammaray_core.dir/probe.cpp.o] Error 1
make[2]: Leaving directory '/tmp/portage/dev-util/gammaray-2.10.0/work/gammaray-2.10.0_build'
make[1]: *** [CMakeFiles/Makefile2:2455: core/CMakeFiles/gammaray_core.dir/all] Error 2
make[1]: Leaving directory '/tmp/portage/dev-util/gammaray-2.10.0/work/gammaray-2.10.0_build'
make: *** [Makefile:163: all] Error 2
```

@a17r ideas about this one?

File: dev-util/gammaray/gammaray-2.10.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,89 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit cmake-utils
```
Comment (juippis) [inline dev-util/gammaray/gammaray-2.10.0.ebuild]: If/When the error gets solved, update to **7** and **inherit cmake**.

**Comment by a17r** [comment]
No, but maybe that was fixed in 2.11.0? https://github.com/KDAB/GammaRay/releases/tag/v2.11.0

**Comment by a17r** [comment]
> No, but maybe that was fixed in 2.11.0? https://github.com/KDAB/GammaRay/releases/tag/v2.11.0

Incidentally was eavesdropping discussion in #plasma today where they mentioned that even 2.11.0 still does not build against Qt 5.14.0. KDAB upstream hopes to have a new release 2.11.1 out soon.

As Qt 5.14.1 stabilisation (and cleanup old) is imminent, only a fixed version of gammaray should be added, as soon as available.

**Comment by a17r** [comment]
https://github.com/KDAB/GammaRay/releases/tag/v2.11.1 is there and mentions Qt 5.14 support in changelog.

Maybe you want to give it another try?

**Comment by a17r** [comment]
So, since I was curious I just moved the current state of ebuild to 2.11.1 and it build fine against Qt 5.14.2, at least with my USE flag settings. So what it would need is a bit of following up upstream changes and getting up to speed with downstream (`EAPI-7`, `cmake-utils.eclass` -> `cmake.eclass`).

**Comment by rossbridger** [comment]
I also tried to move this ebuild to 2.11.1 and it failed to build against Qt 5.15, so it might also need to cherry-pick this upstream commit https://github.com/KDAB/GammaRay/commit/7cd17bc.

**Comment by bansan85** [comment]
I moved the ebuild to 2.11.2. I was able to build with qt 5.15.1 and to attach gammaray with wireshark.

But I noticed an error message:

```
"gammaray-launcher" not found in the expected location(s):
"/usr/bin/..//bin/gammaray-launcher, /usr/bin/..//lib64/gammaray/libexec/gammaray-launcher" 
continuing anyway, hoping for it to be in PATH.
This is likely a setup problem. 
```

qlist doesn't shows gammaray-launcher

**Pull request closed without merging**